projects/ipdg_stokes/post_processing/norms.h

```diff
@@ -9,38 +9,50 @@
 
 #include <lf/mesh/entity.h>
 #include <lf/mesh/mesh.h>
+#include <lf/quad/quad.h>
+#include <lf/uscalfe/uscalfe.h>
 
 #include <functional>
 
+#include <utils.h>
+
 namespace projects::ipdg_stokes {
 
 namespace post_processing {
 
 /**
- * @breif Compute the @f$L^2@f$-norm of a vector valued function over a mesh
- * @param mesh A shared pointer to the mesh over which the function is defined
+ * @brief Compute the @f$L^2@f$-norm of a vector valued function over a mesh
+ * @tparam MF The type of mesh function to take the norm from
+ * @tparam QR_SELECTOR The type of the `qr_selector` parameter
+ * @param mesh The mesh over which the function is defined
  * @param f The function to take the norm from
- * @param quadrule_order The order of the quadrature rule to use to integrate
- * over the elements. Defaults to 50
+ * @param qr_selector A function giving the quadrature rule to use for a given
+ * element
  * @returns The @f$L^2@f$-norm of the function over the mesh
  *
  * This function computes the value of the following formula
  * @f[
  *  ||f||_0^2 = \sum_{K\in\mathcal{M}} \int_K\! ||f(x)||^2 \,\mathrm{d}x
  * @f]
  */
-double L2norm(const std::shared_ptr<const lf::mesh::Mesh> &mesh,
-              const std::function<Eigen::Vector2d(const lf::mesh::Entity &,
-                                                  const Eigen::Vector2d &)> &f,
-              unsigned quadrule_order = 50);
+template <typename MF, typename QR_SELECTOR>
+double L2norm(const std::shared_ptr<const lf::mesh::Mesh> &mesh, const MF &f,
+              const QR_SELECTOR &qr_selector) {
+  return std::sqrt(lf::uscalfe::IntegrateMeshFunction(
+      *mesh, lf::uscalfe::squaredNorm(f), qr_selector));
+}
 
 /**
  * @brief Compute the DG-norm of a vector valued function over a mesh
- * @param mesh A shared pointer to the mesh over which the function is defined
+ * @tparam MF_F The type of mesh function to take the norm from
+ * @tparam MF_GRAD The type of the gradient of the mesh function to take the
+ * norm from
+ * @tparam QR_SELECTOR The type of the `qr_selector` parameter
+ * @param mesh The mesh over which the function is defined
  * @param f The function to take the norm from
  * @param f_grad The gradient of the function to take the norm from
- * @param quadrule_order The order of the quadrature rule to use to integrate
- * over the elements. Defaults to 50
+ * @param qr_selector A function giving the quadrature rule to use for a given
+ * element
  * @returns The @f$H^1@f$-norm of the function over the mesh
```

Comment:
Does it really return the `H^1` norm? It would be nice to put a formula here that explains what the DG Norm is. Otherwise, the implementation is very nice :)